lib/features/autentication/login_page.dart

```diff
@@ -67,12 +60,15 @@ class LoginPageState extends State<LoginPage> {
               ),
               const SizedBox(height: 20),
               ElevatedButton(
-                onPressed: _login,
+                onPressed: () {
+                  final cubit = context.read<LoginCubit>();
+                  _login(context, cubit, emailController, passwordController);
```

Comment:
You can make the onPressed function async, and then you can have the call here without extracted function.